### PR TITLE
Adds multiple file support for Swift zip

### LINKED ISSUE #1985: Allow zip file to contain multiple swift source code files
Allow Swift runtime to accept zip files that contain multiple source code files.

### LINKED ISSUE #2143: bad redirect? in swift build and run generator
@paulcastro : Is the `>2&` here https://github.com/openwhisk/openwhisk/blob/master/core/swift3Action/buildandrecord.py#L71 intended to redirect io streams? As written this will just pipe output to a file named `2`.

### PR DESCRIPTION
**Pull request by paulcastro**
Fixes #1985, #2143.  Allows you to upload multiple Swift source code files in the zip file.  The proxy will now:

1. scan the unzipped source code and concat the code into a single main.swift file. This means the developer only needs to define a main function somewhere. The zip file itself does not need to contain a main.swift.
2. Ignore files with privileged names: `Package.swift`,` _Whisk.swift`, `_WhiskJsonUtils.swift`, which shouldn't be overwritten.  

Note that this does not provide Swift Package Manager support -- this would require allowing arbitrary Package.swift files and full swift builds, which may exceed action limits.

Also Fixes #2143.

### CODE REVIEW

**Comment by paulcastro** [comment]
Blue PG #1187

File: core/swift3Action/buildandrecord.py [hunk 1]
```diff
@@ -48,9 +48,7 @@
 
 ## if found, create build script, otherwise exit with error
 if compileCommand != None and linkCommand != None:
-    print "Generated OpenWhisk Compile command: %s" % compileCommand
-    print "========="
-    print "Generated OpenWhisk Link command: %s" % linkCommand
+    print "Succes, command and link commands found."
```
Comment (rabbah) [inline core/swift3Action/buildandrecord.py]: Success (missing `s`).

File: core/swift3Action/spm-build/_Whisk.swift [hunk 2]
```diff
@@ -184,10 +168,10 @@ class Whisk {
         
     }
     
-    /* 
-        this function is currently unused but ready when we want to switch to using URLSession instead of KituraNet
-    */
- 
+    /*
+     this function is currently unused but ready when we want to switch to using URLSession instead of KituraNet
```
Comment (rabbah) [inline core/swift3Action/spm-build/_Whisk.swift]: warning: nit ahead:
```
/**
 *
 */
```

File: core/swift3Action/spm-build/_Whisk.swift [hunk 3]
```diff
@@ -1,22 +1,6 @@
-/*
```
Comment (rabbah) [inline core/swift3Action/spm-build/_Whisk.swift]: did you intend to remove the headers?
this is likely to come back.

Comment (paulcastro) [inline core/swift3Action/spm-build/_Whisk.swift]: Urg. No that was an accident.  I'll restore

File: core/swift3Action/buildandrecord.py (hunk 1 above)
Comment (paulcastro) [inline core/swift3Action/buildandrecord.py]: Will correct.

File: core/swift3Action/spm-build/_Whisk.swift (hunk 2 above)
Comment (paulcastro) [inline core/swift3Action/spm-build/_Whisk.swift]: That's so Java of you :-)

File: core/swift3Action/swift3runner.py [hunk 4]
```diff
@@ -47,10 +48,18 @@ def epilogue(self, init_message):
             main_function = init_message['main']
         else:
             main_function = 'main'
+        #make sure there is a main.swift file
+        open(DEST_SCRIPT_FILE, 'a').close()
 
         with codecs.open(DEST_SCRIPT_FILE, 'a', 'utf-8') as fp:
+            os.chdir(DEST_SCRIPT_DIR)
+            for file in glob.glob("*.swift"):
```
Comment (bjustin-ibm) [inline core/swift3Action/swift3runner.py]: Oh... my... glob!

File: core/swift3Action/swift3runner.py [hunk 5]
```diff
@@ -47,10 +48,18 @@ def epilogue(self, init_message):
...
+                if file != "Package.swift" and file != "main.swift" and file != "_WhiskJSONUtils.swift" and file != "_Whisk.swift":
```
Comment (bjustin-ibm) [inline core/swift3Action/swift3runner.py]: Not really worth changing now, but if this list gets any longer, you might do something like:

```python
if file not in ["Package.swift", "main.swift", "_WhiskJSONUtils.swift, "_Whisk.swift"] :
```

**Comment by bjustin-ibm** [comment]
LGTM, except there have been non-trivial changes since the last PG. I kicked off PG3 411 with the latest commit.

File: core/swift3Action/buildandrecord.py [hunk 6]
```diff
@@ -52,12 +52,9 @@
...
 
-# if found, create build script, otherwise exit with error
-if compileCommand is not None and linkCommand is not None:
-    print("Generated OpenWhisk Compile command: %s" % compileCommand)
-    print("=========")
-    print("Generated OpenWhisk Link command: %s" % linkCommand)
-
+## if found, create build script, otherwise exit with error
+if compileCommand != None and linkCommand != None:
+    print "Success, command and link commands found."
```
Comment (bjustin-ibm) [inline core/swift3Action/buildandrecord.py]: @paulcastro syntax error. Needs parens?

**Comment by akrabat** [comment]
What's the status on this?

**Comment by rabbah** [comment]
I rebased this to master to pick up a performance fix in the JSON serialization and resolve conflict. Also fixed one line for python 3 syntax and removed trailing white spaces.

**Comment by rabbah** [comment]
pg2/1445 🏃

**Comment by bjustin-ibm** [comment]
pg2 #1450 🏃

**Comment by rabbah** [comment]
@bjustin-ibm pgs 2/1447 and 2/1448 are ok - I'm looking into those recurring three failure which are not related. I've had enough of them.

**Comment by rabbah** [comment]
I just realized we didn't add a test specific for this change.

**Comment by paulcastro** [comment]
@rabbah thanks for saying "we" instead of @paulcastro :-D

I forgot that important step.  Let me add a test.

**Comment by paulcastro** [comment]
New test succeed locally.  PG #530  🏃

**Comment by paulcastro** [comment]
Added fix for #2143 to this since its a single typo fix.

**Comment by rabbah** [comment]
pg1/1408 approved.

**Pull request merged**